### ISSUE TITLE
do we need a config server?  can we use webpack.devServer.setup function instead? multiple ports in dev are causing slight friction...

### ISSUE THREAD

**Issue by rileylnapier** [nozzle/react-static#163]
hi!  me again :) i'm loving the library and obvy have an idea or two on what i want to build

so i'm looking to host a dev version of an app i built on react-static on aws.  my company has a docker/jenkins flow to ecs that allows for one port to be exposed via load balancer so using react-static in this case is giving us trouble.

i also tried using heroku... but it doesn't like multiple ports either :(

my ultimate goal is to create endpoints on a dev-server that can auto rebuild a specific page via CMS webhook, or even rebuild all the pages... 

so say if i edit a product name on the CMS, it will ping my dev server, use react-static to rebuild that one page... then push it off to a s3 bucket.

at the same time i want to use "react-static start" command to be a dev-preview of the entire website before the we deploy to s3.

# alternative i can think of:
instead of hosting react-static dev server on aws, just configure the webhooks from CMS to something like circleci and push dev versions of the site to alternate s3 buckets.  the build process is kinda slow so i would rather just have a dev server that always pulls the data dynamically

# possible solutions
we can extend webpack devServer 
https://webpack.js.org/configuration/dev-server/#devserver-setup

and add the routes directly to the dev server?  we could prefix routes with something like __config__  or __static__ to prevent collision?

thoughts?

react-static/src/webpack/index.js
```
const devServerConfig = {
  hot: true,
  disableHostCheck: true,
  contentBase: config.paths.DIST,
  publicPath: '/',
  historyApiFallback: true,
  compress: false,
  quiet: true,
  watchOptions: {
    ignored: /node_modules/,
  },
  ...config.devServer,
  setup: (app) => {
    app.get('/__config__/siteProps', async (req, res, next) => {}
    app.get('/__config__/getRoutes', async (req, res, next) => {}

    if (config.devServer && config.devServer.setup) {
      config.devServer.setup(app);
    }
  },
  port,
  host,
}
```

i can help with PR too if you need?

**Comment by rileylnapier** [comment]
i made this:

https://github.com/nozzle/react-static/pull/164

let me know if there are any requirements I am missing for a PR

**Comment by tannerlinsley** [comment]
Closed with #164 :) Great work!
